pkg/provider/github/parse_payload.go

```diff
@@ -259,6 +259,15 @@ func (v *Provider) processEvent(ctx context.Context, event *info.Event, eventInt
 		if gitEvent.GetRepo() == nil {
 			return nil, errors.New("error parsing payload the repository should not be nil")
 		}
+
+		// When a branch is deleted via repository UI, a push event is triggered,
+		// hence returning an error from here to avoid further processing
+		if gitEvent.After != nil {
+			if isZeroSHA(*gitEvent.After) {
+				return nil, fmt.Errorf("branch %s has been deleted, exiting", gitEvent.GetRef())
+			}
+		}
```

Comment:
```suggestion
		if isZeroSHA(*gitEvent.After) && *gitEvent.GetHeadCommit() == nil {
			return nil, fmt.Errorf("the branch %s has been deleted, but the \"Delete Branch\" event is not supported", gitEvent.GetRef())
		}
```

I don't think that `gitEvent.After` would be nil in any case?

Reply:
(but that's not a problem if checked)